pyiron_atomistics/__init__.py

```diff
@@ -24,51 +24,52 @@
-JOB_CLASS_DICT["ART"] = "pyiron_atomistics.interactive.activation_relaxation_technique"
-JOB_CLASS_DICT["AtomisticExampleJob"] = "pyiron_atomistics.testing.randomatomistic"
-JOB_CLASS_DICT["Calphy"] = "pyiron_atomistics.calphy.job"
-JOB_CLASS_DICT[
-    "ConvEncutParallel"
-] = "pyiron_atomistics.dft.master.convergence_encut_parallel"
-JOB_CLASS_DICT[
-    "ConvEncutSerial"
-] = "pyiron_atomistics.dft.master.convergence_encut_serial"
-JOB_CLASS_DICT[
-    "ConvergenceVolume"
-] = "pyiron_atomistics.atomistics.master.convergence_volume"
-JOB_CLASS_DICT[
-    "ConvKpointParallel"
-] = "pyiron_atomistics.dft.master.convergence_kpoint_parallel"
-JOB_CLASS_DICT["ElasticTensor"] = "pyiron_atomistics.atomistics.master.elastic"
-JOB_CLASS_DICT["ExampleJob"] = "pyiron_atomistics.testing.randomatomistic"
-JOB_CLASS_DICT["Gaussian"] = "pyiron_atomistics.gaussian.gaussian"
-JOB_CLASS_DICT["Gpaw"] = "pyiron_atomistics.gpaw.gpaw"
-JOB_CLASS_DICT["HessianJob"] = "pyiron_atomistics.thermodynamics.hessian"
-JOB_CLASS_DICT["Lammps"] = "pyiron_atomistics.lammps.lammps"
-JOB_CLASS_DICT["MapMaster"] = "pyiron_atomistics.atomistics.master.parallel"
-JOB_CLASS_DICT["Murnaghan"] = "pyiron_atomistics.atomistics.master.murnaghan"
-JOB_CLASS_DICT["MurnaghanDFT"] = "pyiron_atomistics.dft.master.murnaghan_dft"
-JOB_CLASS_DICT["PhonopyJob"] = "pyiron_atomistics.atomistics.master.phonopy"
-JOB_CLASS_DICT["QuasiHarmonicJob"] = "pyiron_atomistics.atomistics.master.quasi"
-JOB_CLASS_DICT["QuasiNewton"] = "pyiron_atomistics.interactive.quasi_newton"
-JOB_CLASS_DICT["ScipyMinimizer"] = "pyiron_atomistics.interactive.scipy_minimizer"
-JOB_CLASS_DICT["SerialMaster"] = "pyiron_atomistics.atomistics.master.serial"
-JOB_CLASS_DICT["Sphinx"] = "pyiron_atomistics.sphinx.sphinx"
-JOB_CLASS_DICT[
-    "StructureContainer"
-] = "pyiron_atomistics.atomistics.job.structurecontainer"
-JOB_CLASS_DICT["StructureListMaster"] = "pyiron_atomistics.atomistics.master.structure"
-JOB_CLASS_DICT["SQSJob"] = "pyiron_atomistics.atomistics.job.sqs"
-JOB_CLASS_DICT["SQSMaster"] = "pyiron_atomistics.atomistics.master.sqsmaster"
-JOB_CLASS_DICT["SxDynMat"] = "pyiron_atomistics.thermodynamics.sxphonons"
-JOB_CLASS_DICT["SxExtOptInteractive"] = "pyiron_atomistics.interactive.sxextoptint"
-JOB_CLASS_DICT["SxHarmPotTst"] = "pyiron_atomistics.thermodynamics.sxphonons"
-JOB_CLASS_DICT["SxPhonons"] = "pyiron_atomistics.thermodynamics.sxphonons"
-JOB_CLASS_DICT["SxUniqDispl"] = "pyiron_atomistics.thermodynamics.sxphonons"
-JOB_CLASS_DICT["TableJob"] = "pyiron_atomistics.table.datamining"
-JOB_CLASS_DICT["Vasp"] = "pyiron_atomistics.vasp.vasp"
-JOB_CLASS_DICT["VaspMetadyn"] = "pyiron_atomistics.vasp.metadyn"
-JOB_CLASS_DICT["VaspSol"] = "pyiron_atomistics.vasp.vaspsol"
+JobType.register("pyiron_atomistics.interactive.activation_relaxation_technique", "ART")
+JobType.register("pyiron_atomistics.testing.randomatomistic", "AtomisticExampleJob")
+JobType.register("pyiron_atomistics.calphy.job", "Calphy")
```

Comment:
I still wonder if this is any more readable than using the `JOB_CLASS_DICT`.

Reply:
In my opinion it is slightly more clear what is being done. `JobType.register` tells that you are registering a job type. Also the arguments of that function tell that they are `job_class_or_module_str` and `job_name`.  This, you cannot guess directly from `JOB_CLASS_DICT['job_name'] = "job.module:` but we rely on further documentation. Furthermore, changes to the `JobType` can be tracked and name clashes are reported. Once we have another `TableJob` in e.g. experimental, this framework also allows to add some hidden `_extra_require` dict to hold additional modules to be imported (i.e. please import `datamining.TableJob` from atomistics and from experimental and return the standard `TableJob` from base).